Source/WebInspectorUI/UserInterface/Models/CSSProperty.js

```diff
@@ -503,6 +548,40 @@ WI.CSSProperty = class CSSProperty extends WI.Object
 
     // Private
 
+    _updateName(name)
+    {
+        if (name === this._name)
+            return;
+
+        let changeCount = (propertyName, delta) => {
+            if (!propertyName || this._implicit || this._anonymous || !this._enabled)
+                return;
+
+            let oldCount = WI.CSSProperty._cachedNameCounts[propertyName];
+
+            // Allow property counts to be updated if the property name has already been counted before.
+            // This can happen when inspecting a device that has different CSS properties enabled.
+            if (isNaN(oldCount) && !WI.cssManager.propertyNameCompletions.isValidPropertyName(propertyName))
+                return;
+
+            let newCount = Math.max(0, (oldCount || 0) + delta);
+            WI.CSSProperty._cachedNameCounts[propertyName] = newCount;
+
+            WI.objectStores.cssPropertyNameCounts.get(propertyName).then(async (count) => {
+                count = Math.max(0, (count || oldCount || 0) + delta);
+                await WI.objectStores.cssPropertyNameCounts.put(count, propertyName);
+                WI.CSSProperty._cachedNameCounts[propertyName] = count;
```

Comment:
Since this is inside an `async` function, is it actually 100% safe for us to not wait on this function to return before returning from `changeCount`? We call `changeCount` twice below, so I wonder what happens if putting something in the object store takes a bit longer and we then update the cached counter after the next invocation has already grabbed it to use as the old count.

Reply:
ah yeah this should probably happen before the `await`

Just as a FYI, I'm pretty sure IndexedDB operations (which is what `WI.ObjectStore` is built on top of) are guaranteed to never happen out of order, so yes there may be a brief period where the `WI.CSSProperty._cachedNameCounts` is out of sync with `WI.objectStores.cssPropertyNameCounts`, but this logic should make sure that after all edits and IndexedDB operations both `WI.CSSProperty._cachedNameCounts` and `WI.objectStores.cssPropertyNameCounts` have the same value (for a given property).